### ISSUE TITLE
topoaa does not check for similar chainIDs (and does not allow to define those in the config file)

### ISSUE THREAD

**Issue by amjjbonvin** [haddocking/haddock3#741]
**Describe the bug**

Buy default haddock3 takes the chainID/segID from the input PDB files. When two files are given with the same chainID, there is not check done and this will cause the next steps to fail (problem reported by an user). Basically the topology (psf) of the complex will only contain one chain (or some weird combination, but only with chain A). 

In principle, in the parameters (yaml file) we allow to define manually the segIDs which should allow to correct that:

```
(haddock3) bash-3.2$ haddock3-cfg -m topoaa

...
[topoaa.mol1]
prot_segid = 'A'  # $minchars 0 / $maxchars 4 / $title Segment ID / $group molecule

```

Specifying different `port_segid` values for the two chains in the config file is however not resolving the problem as those are ignored.

**To Reproduce**
Steps to reproduce the behavior, if relevant, provide the input that triggered the error:

Give twice the same molecule (but different filenames) for the docking in the config file and try some simple docking scenario.

Try to specify the prot_segid for each molecule in the config file, e.g.:

```
[topoaa]
[topoaa.mol1]
prot_segid = 'A'
[topoaa.mol2]
prot_segid = 'B'
```

**Expected behavior**

Ideally, haddock3 should check that the various molecules given as input don't have the same chainIDs.

Or, manually specifying the chainIDs should overwrite the chainIDs/segIDs from the input proteins.

Simple dataset to reproduce the problem:

[test-haddock3.tgz](https://github.com/haddocking/haddock3/files/13319732/test-haddock3.tgz)

**Comment by mgiulini** [comment]
hi @amjjbonvin , I digged a bit into the bug. Indeed it seems that the `prot_segid` parameter is simply passed to the topoaa CNS code, without being actually used by CNS (although the description says that it's being used at the CNS level). A `prot_segid_1` parameter is also defined in the CNS routine: in the .inp file there's the following line:

`
!do (segid = $prot_segid_1) (all)
`

but re-running topoaa with `eval ($prot_segid_1="B")` does not change the outcome..

I don't have a preference for any of the two solutions, but probably we should try to be consistent with the server and overwrite the chain ID (if specified). The problem here is that if topoaa.mol2 has already two chains (say A and B) overwriting B to the whole PDB would cause the topology to fail. But I guess that expecting a PDB with a single chain is fair (for non scoring purposes of course). @rvhonorato what do you say?

**Comment by amjjbonvin** [comment]
For scoring indeed we should keep the chainID as they are in the  file.

But for a docking scenario where two or more topologies are combined, the segIDs must be different.

If we uncomment the line as you suggested it would work for a docking scenario, but not for refinement and/or scoring.

We should also then remove the prot_segid parameters from the yaml file of topoaa since they are no longer used.

Ideally, when defining more than one molecule in topoaa, there should be a check that their chainIDs/segIDs are different.

**Comment by mgiulini** [comment]
> For scoring indeed we should keep the chainID as they are in the file. But for a docking scenario where two or more topologies are combined, the segIDs must be different. If we uncomment the line as you suggested it would work for a docking scenario, but not for refinement and/or scoring. We should also then remove the prot_segid parameters from the yaml file of topoaa since they are no longer used. Ideally, when defining more than one molecule in topoaa, there should be a check that their chainIDs/segIDs are different.

placing a simple check at the topoaa level is likely the easiest solution indeed.

**Comment by rvhonorato** [comment]
I'd say we just take whatever is in the PDB, internally set it as the segid and remove this parameter alltogether.

**Comment by amjjbonvin** [comment]
Yes - but we need to check that when giving more than one molecule, the segIDs/chainIDs in those PDB file must be different, otherwise topoaa will fail.

**Comment by rvhonorato** [comment]
The modules should not data-validation, isn't there a whole subroutine just for this?

**Comment by amjjbonvin** [comment]
This particular validation is very specific to the topoaa module.

**Comment by rvhonorato** [comment]
There is no implementation of data validation in the modules, to handle this we need to go into the esoteric parts of the code to handle it - so much easier and doable to do it in the pre-processing part

**Comment by mgiulini** [comment]
hi there, I looked at this issue and found a possible fix, namely a check during the preparation of the CNS jobs: there (see https://github.com/haddocking/haddock3/blob/6389a20bfeb41c931f152213225419e658a5c017/src/haddock/libs/libcns.py#L332

chain IDs and segids are already retrieved, so it's quite simple to check whether they are repeated across multiple pdb files (for non-scoring modules).

What do you say? it's impossible to do this check at the preprocessing stage (the files could enter a scoring workflow), and I would not change the chain ID at topoaa..adding a check here makes the error quite clear.
